krun/util.py

```diff
@@ -486,7 +486,18 @@ def _do_reboot(platform):
         os.execv(args[0], args)  # replace myself
         assert False  # unreachable
     else:
-        subprocess.call(platform.get_reboot_cmd())
+        rc = subprocess.call(platform.get_reboot_cmd())
+        if rc != 0:
+            fatal("Failed to reboot with: %s" % platform.get_reboot_cmd())
+        else:
+            debug("hard exit")
+            # Really exit right now. No more Python code should execute. A regular
+            # sys.exit() would raise a 'SystemExit' which unwinds the stack
+            # including running 'finally' blocks, which is undesirable if you want
+            # to reboot. This is especially acute in the case of reboot, since the
+            # amount of 'finally' code that executes is racy depending on how quick
+            # the system reboots.
```

Comment:
OK I eventually understood this comment, but it's really long-winded. Can we shorten and clarify?